src/target.rs

```diff
@@ -75,6 +75,8 @@ impl Target {
             && (os == "linux" || os == "freebsd" || os == "dragonfly" || os == "netbsd")
         {
             lines.push(format!("-Wl,-soname,lib{}.so.{}", lib_name, major));
+        } else if  os == "haiku") {
```

Comment:
while at it please fold it in the above branch please.

Reply:
I'll do another round to add it to the above statement today, will get back to that later.

Reply:
Also please remember to use `cargo fmt` :)

Reply:
> Also please remember to use `cargo fmt` :)

Not sure what you mean by that (sorry not a developer here), looking at "cargo fmt" it would be a separate package (not available in Haiku atm)?
If I look at other recipes around at repology I don't see it being used there?

Reply:
`cargo fmt` is part of [rustfmt](https://github.com/rust-lang/rustfmt) it is enforced at CI level in many project, including cargo-c.

Reply:
I'll have a look at that later, atm rebuilding with some changes to this PR (specially in `target.rs`)

Reply:
Poking @nielx as he is our rust master :)

Reply:
My bad, seems it's part of our rust package ... https://github.com/haikuports/haikuports/blob/master/dev-lang/rust_bin/rust_bin-1.60.0.recipe#L39